converter.go

```diff
@@ -1084,7 +1084,7 @@ func arrowToRecord(ctx context.Context, record arrow.Record, pool memory.Allocat
 							ar = arrow.Timestamp(ts.UnixNano())
 							// in case of overflow in arrow timestamp return error
 							// this could only happen for nanosecond case
-							if ts.Year() != ar.ToTime(arrow.Nanosecond).Year() {
+							if ts.UTC().Year() != ar.ToTime(arrow.Nanosecond).Year() {
```

Comment:
Great catch indeed! Can you add a test to confirm it doesn't work without the fix and works with it?

Reply:
@sfc-gh-pfus done, using the existing test case, but changed to 1/1. The test failed before the fix.